Controllers/ProjectControllers/Admin/UserController.php

```diff
@@ -0,0 +1,204 @@
+                            }
+                        } else {
+                            if (!$this->userModel->newUser($params[0], $correctResult)) {
+                                $this->adminLocation('/user/registration');
+                            } else {
+                                $this->adminHomeLocation();
+                            }
+                        }
+                    } else {
+                        foreach ($correctResult as $key => $value) {
+                            if (!$value) {
+                                $this->msgModel->setMsg($this->msgModel->notCorrectData($key));
+                            }
+                        }
+                        $this->adminLocation('/user/registration');
+                    }
+                } else {
+                    foreach ($emptyResult as $key => $value) {
+                        if (!$value) {
+                            $this->msgModel->setMsg($this->msgModel->emptyDataMsg($key));
+                        }
+                    }
+                    $this->adminLocation('/user/registration');
+                }
+            } else {
+                $this->adminHomeLocation();
+            }
+        } else {
+            $this->homeLocationByCustomerType();
+        }
+    }
+
+    /**
+     * Logout action
+     *
+     * @return void
+     */
+    public function logoutAction(): void
+    {
+        if ($this->userModel->isSigned()) {
+            if ($this->userModel->isAdmin()) {
+                $this->userModel->logout();
+                $this->adminHomeLocation();
+            } else {
+                $this->homeLocation();
+            }
+        } else {
+            $this->adminHomeLocation();
+        }
+    }
+
+    /**
+     * if user is logged sends to homeLocation depending on if it is a simple user or admin user
+     */
+    public function homeLocationByCustomerType(): void
+    {
+        if ($this->userModel->isAdmin()) {
+            $this->adminHomeLocation();
+        } else {
+            $this->homeLocation();
+        }
+    }
+
+    /**
+     * Render form for authorization
+     *
+     * @return void
+     */
+    public function authorizationAction(): void
+    {
+        if (!$this->userModel->isSigned()) {
+            $options = $this->userView->getOptions('Авторизация', 'admin/login.phtml');
+            $this->userView->render($options);
+        } else {
+            $this->homeLocationByCustomerType();
+        }
+    }
+
+    /**
+     * Login action
+     *
+     * @return void
+     */
+    public function loginAction(array $params = null): void
+    {
+        if (!$this->userModel->isSigned()) {
+            if ($this->postInfo->isPost()) {
+                $emptyResult = $this->dataValidator->emptyCheck($this->postInfo->getPostData(), 'login');
+                if (!in_array(false, $emptyResult)) {
+                    if (!$this->userModel->login($params[0], $emptyResult)) {
```

Comment:
**P2** - I can see the variable $params can be null. Don't you think checking for a first element from array when variable might be null would lead to a php parse error?